### DISCUSSION TITLE
shader settings and color profiles

### DISCUSSION THREAD

**Discussion by TiborMitovics** [google/model-viewer#2994] [Q&A] [answered]
Is the model viewer using any kind of color profile or post processing in the shader?

I am currently producing 3D models with textures/materials for a client that is using your model-viewer. My Software is Adobe Substance 3D Painter version 7.2.0. I noticed huge differences in lighting contrast and color display between the model viewer and Substance Painter. 

My Substance Painter settngs:
shader: PBR-metal-rough (openGL)
Colorspace: sRGB
color profile (color LUT): sRGBf
HDR exposure in Substance Painter: 2.5
HDR exposure in the model-viewer: 5 (achieved by using the inspect tool from google chrome, this value is set by the client)

For both the model-viewer and Substance Painter I am using the same HDR with the same z rotation.
Some colors are more saturated and some less, especially the yellow/orange tones. Attached are the HDR, the LUT from Substance Painter and 2 screenshots from Painter and the model-viewer.

I will also post this thread in the substance forums to get suggestions from the other side aswell.

Thanks a lot in advance!

[3D Viewer HDR.
![ColorsSubstancePainter](https://user-images.githubusercontent.com/95038106/143465922-3a90a39b-f336-4fa9-8a1d-0e2b8b8ca849.png)
zip](https://github.com/google/model-viewer/files/7603874/3D.Viewer.HDR.zip)
![ColorsModelViewer](https://user-images.githubusercontent.com/95038106/143465905-f466278c-2266-4eb3-8240-a152d0bf3e72.png)
[sRGBf.zip](https://github.com/google/model-viewer/files/7603888/sRGBf.zip)

.

**Comment by elalish** [comment, marked as answer]
This  kind of difference is due to tone mapping and exposure. You can control exposure with the MV attribute, but that probably won't be enough. We use ACES tone mapping, so you'll need to set up Substance Painter to use the same if you want to compare results. This is an area that is a bit lacking in specification for glTF and  results in most of the color difference  between  renderers, so  we're working on an extension to help  standardize it. If you have  strong opinions  on tone mapping functions, please put them here and we'll do our best  to  build something that meets everyone's needs.

**Reply by TiborMitovics** [reply]
Thanks a lot, this helps a lot already to understand what is going on.
As a suggestion you could add this information about the ACES tone mapping into the documentaion so other content creators can easily adapt to it.
kind regards

**Reply by fredericocm** [reply]
Hello @TiborMitovics ! Did you manage to set up Substance Painter to match the colors from the model viewer? If so, could you share this information? I'm going through the same issue

**Reply by TROVIC93** [reply]
In the latest version of Substance Painter, when creating a new project select "OpenColorIO" under Color Management and ACES 1.2 under "OpenColorIO configuration" (screenshot attached)
![SP_ACES](https://user-images.githubusercontent.com/47480985/155345165-6e26d4f5-f503-4231-bfd8-0a2dd008269b.png)
Make sure you are using the same HDRI in the model viewer and SP. You simply have to adjust the HDRI exposure in Substance to match it to the viewer. This value might change depending on your HDRI.
I managed to get a 98% match (estimated)
See the image below. the parts on the left in frame are screenshots from Substance overlayed onto the viewer

![Comparison](https://user-images.githubusercontent.com/47480985/155345195-a64ca254-f3c5-4712-8240-0481dbbaab3d.png)

**Reply by fredericocm** [reply]
@TROVIC93 thank you so much for sharing that info! My only question is: how did you get the rotation and exposure values from model viewer's HDR image? I tried used the ispect tool, but I didn't find those info (I'm not much familiar with this tool actually)

**Reply by TROVIC93** [reply]
The exposure 5 setting is actually a requirement by a client for this specific HDRI, so it is not a must if you use your own HDRI.
The model viewer has a exposure slider but it caps at 2. Only if you want to go higher you have to use the inspect tool
See screenshot below:
![inspect tool](https://user-images.githubusercontent.com/47480985/155508715-8b960f0a-fef2-4102-9c40-5be103ce5e37.png)

To get the same rotation is quite easy: both the viewer and Substance have an option to make the HDRI visible in the background so you can simply allign them.
In the viewer it is called "use Environment as Skybox" and in Substance you have to turn up the Environment Opacity in the display settings.
For my specific case i had to rotate the HDRI 180 degrees to match the rotation in the viewer

**Reply by fredericocm** [reply]
Got it, thanks! I tried all the settings and I saw a big improvement in color matching between viewer and Painter, they helped a lot!. But I still didn't get such an accurate result as the one you got. The yellow tones especially are still looking different. They are more orange in Painter. But I'm playing with tone mapping and other settings on Painter to see if I get that right
